### ISSUE TITLE
feat(language): implement autoregressive text generation

### ISSUE THREAD

**Issue by leandronsp** [leandronsp/aspirina#7]
## Overview

Implement autoregressive generation in `language/generate.rs`. The model generates text token by token: predict next token, append it, predict again. Temperature sampling controls creativity. This is the grand finale. The model "speaks".

## Problem Statement

A trained model can compute next-token probabilities, but we need the generation loop: sample a token from the distribution, feed it back as input, repeat until we hit a stop condition or max length. Temperature controls the trade-off between deterministic (always pick the highest probability) and creative (sample from the full distribution).

## Current State

- Trained nano-LLM produces logits over vocabulary (from issue #6)
- Tokenizer can decode token IDs back to text (from issue #2)
- No generation/inference code exists

## Requirements

- [ ] R-1: Implement greedy decoding: always pick the highest-probability token
- [ ] R-2: Implement temperature sampling: divide logits by temperature before softmax, then sample from the distribution
- [ ] R-3: Implement top-k sampling: zero out all but the k highest logits before softmax
- [ ] R-4: Implement the generation loop: prompt -> encode -> generate N tokens -> decode -> text output
- [ ] R-5: Support configurable max_tokens and temperature parameters
- [ ] R-6: Demonstrate: give the model a prompt, watch it generate text character by character

## Technical Approach

```
language/src/
  generate.rs     # GenerationConfig, generate() function
```

Generation loop:
```
tokens = tokenizer.encode(prompt)
for _ in 0..max_tokens:
  x = model.forward(tokens)              # full forward pass
  logits = x[-1]                         # last position only
  logits = logits / temperature           # temperature scaling
  probs = softmax(logits)
  next_token = sample(probs)             # weighted random choice
  tokens.push(next_token)
print(tokenizer.decode(tokens))
```

Temperature:
- 0.0 = greedy (argmax)
- 1.0 = sample from learned distribution
- >1.0 = more random/creative

## Acceptance Criteria

- Given a trained model and temperature=0, when I generate from the same prompt twice, then I get identical output
- Given temperature=1.0, when I generate from the same prompt twice, then outputs differ (probabilistic sampling)
- Given a prompt "ROMEO:", when I generate 100 tokens, then the output resembles Shakespearean text patterns (even if garbled)
- Given max_tokens=50, when I generate, then output is exactly 50 tokens after the prompt

## Out of Scope

- KV-cache for fast inference (recomputes full sequence each step)
- Beam search
- Streaming output
- Interactive chat/REPL mode
